### PR TITLE
Use fork of re-frame with react-native bugfixes

### PR DESCRIPTION
**Pull request by yenda**
It  would be best to have the artefact deployed somewhere that we control, possible with automated build from our repo on jenkins. For now I used my fork and clojar so that QA can test it.

status: ready

### CODE REVIEW

**Review by janherich — APPROVED** [review]

**Comment by dmitryn** [comment]
Crash log with debug lines marked with `[INIT]` prefix:
```
02-14 11:45:27.205  5674  5722 D ReactNativeJS: DEBUG [status-im.data-store.realm.schemas.account.v21.core:62] - migrating v21 account database:  [object Realm] [object Realm]
02-14 11:45:27.464   766  1450 I WifiService: getConnectionInfo uid=10127
02-14 11:45:27.557 26186 26228 E NotifUtils: null sender iterating over styledSenders
02-14 11:45:27.652  5674  5722 D ReactNativeJS: DEBUG [status-im.android.core:84] - [INIT] android.core/init
02-14 11:45:27.656  5674  5722 D ReactNativeJS: DEBUG [status-im.native-module.impl.module:29] - :store-call function () {return status_im.native_module.impl.module.status.setSoftInputMode(e);}
02-14 11:45:27.658  5674  5722 D ReactNativeJS: DEBUG [status-im.android.core:86] - [INIT] status/set-soft-input-mode
02-14 11:45:27.661  5674  5722 D ReactNativeJS: DEBUG [status-im.android.core:88] - [INIT] init-back-button-handler!
02-14 11:45:27.727  5674  5722 I ReactNativeJS: Running application "StatusIm" with appParams: {"rootTag":1}. __DEV__ === false, development-level warning are OFF, performance optimizations are ON
02-14 11:45:27.985   431   431 D SurfaceFlinger: duplicate layer name: changing im.status.ethereum/im.status.ethereum.MainActivity to im.status.ethereum/im.status.ethereum.MainActivity#2
02-14 11:45:28.027  5674  5723 D StatusModule: parseJail chatId:console
02-14 11:45:28.070  5674  5723 F libc    : Fatal signal 11 (SIGSEGV), code 1, fault addr 0x4 in tid 5723 (mqt_native_modu), pid 5674 (status.ethereum)
```

Not sure why `parseJail chatId:console` called, but `initJail` was not even there.

As an example, normal app run log:
```
02-13 15:19:18.986  2325  2369 D ReactNativeJS: DEBUG [status-im.android.core:84] - [INIT] android.core/init
02-13 15:19:18.988  2325  2370 D StatusModule: setSoftInputMode
02-13 15:19:18.989  2325  2369 D ReactNativeJS: DEBUG [status-im.android.core:86] - [INIT] status/set-soft-input-mode
02-13 15:19:18.991  2325  2369 D ReactNativeJS: DEBUG [status-im.android.core:88] - [INIT] init-back-button-handler!
02-13 15:19:18.992  2325  2369 D ReactNativeJS: DEBUG [status-im.core:17] - [INIT] log/set-level!
02-13 15:19:18.995  2325  2369 D ReactNativeJS: DEBUG [status-im.core:19] - [INIT] error-handler/register-exception-handler!
02-13 15:19:18.999  2325  2369 D ReactNativeJS: DEBUG [status-im.core:21] - [INIT] status/init-jail
02-13 15:19:18.999  2325  2370 D StatusModule: initJail
02-13 15:19:19.008  2325  2369 D ReactNativeJS: DEBUG [status-im.core:23] - [INIT] data-store/init
02-13 15:19:19.010  2325  2369 D ReactNativeJS: DEBUG [status-im.core:25] - [INIT] .registerComponent
02-13 15:19:19.013  2325  2369 D ReactNativeJS: DEBUG [status-im.utils.handlers:29] - Handling re-frame event:  :initialize-app
```

Full crash logs here: https://gist.github.com/dmitryn/8400f417ecdb3d8b421e96474e9ccf26#file-nexus5x-crash-log

<blockquote><div> Gist</div><div><strong><a href="https://gist.github.com/dmitryn/8400f417ecdb3d8b421e96474e9ccf26">re-frame fork crash log https://i.diawi.com/aJkxCm</a></strong></div><div>re-frame fork crash log https://i.diawi.com/aJkxCm</div></blockquote>

**Comment by martinklepsch** [comment]
@yenda consider also opening an issue against re-frame which details your changes. In my experience the re-frame folks are pretty open to improving stuff. I'm sure you don't want to maintain your fork forever 🙂

**Comment by dmitryn** [comment]
@yenda 
App start log when you did app upgrade not fresh install :

```
02-14 13:31:28.105 13266 13312 I ReactNativeJS: There was an error collecting entropy from the browser:
02-14 13:31:28.111 13266 13312 I ReactNativeJS: { [TypeError: undefined is not an object (evaluating 'o.random')] line: 660, column: 465, sourceURL: 'index.android.bundle' }
02-14 13:31:28.465   766  1039 I WifiService: getConnectionInfo uid=10130
02-14 13:31:29.522   766 17323 I WifiService: getConnectionInfo uid=10130
02-14 13:31:30.499   766  7638 I WifiService: getConnectionInfo uid=10130
02-14 13:31:31.505   766  9365 I WifiService: getConnectionInfo uid=10130
02-14 13:31:31.552 13266 13312 D ReactNativeJS: DEBUG [status-im.android.core:84] - [INIT] android.core/init
02-14 13:31:31.565 13266 13312 D ReactNativeJS: DEBUG [status-im.native-module.impl.module:29] - :store-call function () {return status_im.native_module.impl.module.status.setSoftInputMode(e);}
02-14 13:31:31.571 13266 13312 D ReactNativeJS: DEBUG [status-im.android.core:86] - [INIT] status/set-soft-input-mode
02-14 13:31:31.577 13266 13312 D ReactNativeJS: DEBUG [status-im.android.core:88] - [INIT] init-back-button-handler!
02-14 13:31:31.693 13266 13312 I ReactNativeJS: Running application "StatusIm" with appParams: {"rootTag":1}. __DEV__ === false, development-level warning are OFF, performance optimizations are ON
02-14 13:31:31.703   431   431 D SurfaceFlinger: duplicate layer name: changing im.status.ethereum/im.status.ethereum.MainActivity to im.status.ethereum/im.status.ethereum.MainActivity#2
02-14 13:31:32.315 13266 13271 I zygote  : Do partial code cache collection, code=63KB, data=45KB
02-14 13:31:32.317 13266 13271 I zygote  : After code cache collection, code=62KB, data=44KB
02-14 13:31:32.317 13266 13271 I zygote  : Increasing code cache capacity to 256KB
02-14 13:31:32.518   766  1031 I WifiService: getConnectionInfo uid=10130
02-14 13:31:32.542 13266 13312 W ReactNativeJS: MenuContext is deprecated and it might be removed in future releases, use MenuProvider instead.
02-14 13:31:32.937 13266 13315 D StatusModule: setSoftInputMode
02-14 13:31:32.942 13266 13315 D StatusModule: initJail
02-14 13:31:33.005 13266 13266 I zygote  : Deoptimizing void com.facebook.react.uimanager.UIViewOperationQueue$1.run() due to JIT inline cache
02-14 13:31:33.074 13266 13276 I zygote  : Background concurrent copying GC freed 23398(1401KB) AllocSpace objects, 6(128KB) LOS objects, 49% free, 3MB/7MB, paused 250us total 113.487ms
02-14 13:31:33.152 13266 13315 D StatusModule: parseJail chatId:console
02-14 13:31:33.385   564  6410 D audio_route: Apply path: speaker-protected
02-14 13:31:33.391   564  6410 D audio_hw_primary: enable_snd_device: snd_device(95: vi-feedback)
02-14 13:31:33.391   564  6410 D audio_route: Apply path: vi-feedback
02-14 13:31:33.391   564  6410 D audio_hw_primary: enable_audio_route: usecase(24) apply and update mixer path: spkr-vi-record
02-14 13:31:33.391   564  6410 D audio_route: Apply path: spkr-vi-record
02-14 13:31:33.417   564  6410 D audio_hw_primary: enable_audio_route: usecase(1) apply and update mixer path: low-latency-playback
02-14 13:31:33.417   564  6410 D audio_route: Apply path: low-latency-playback
02-14 13:31:33.428 13266 13315 D StatusModule: {"result":"{\"commands\":{\"faucet,50\":{\"name\":\"faucet\",\"title\":\"Faucet\",\"description\":\"Get some ETH\",\"has-handler\":true,\"async-handler\":false,\"color\":\"#7099e6\",\"params\":[{\"name\":\"url\",\"type\":\"text\",\"placeholder\":\"Faucet URL\"}],\"scope\":[\"personal-chats\",\"registered\",\"dapps\"],\"scope-bitmask\":50},\"debug,50\":{\"name\":\"debug\",\"title\":\"Debug mode\",\"description\":\"Starts\/stops a debug mode\",\"has-handler\":true,\"async-handler\":false,\"color\":\"#7099e6\",\"params\":[{\"name\":\"mode\",\"type\":\"text\"}],\"scope\":[\"personal-chats\",\"registered\",\"dapps\"],\"scope-bitmask\":50}},\"responses\":{\"password,42\":{\"name\":\"password\",\"description\":\"Password\",\"has-handler\":true,\"async-handler\":false,\"color\":\"#7099e6\",\"icon\":\"lock_white\",\"params\":[{\"name\":\"password\",\"type\":\"password\",\"placeholder\":\"Type your password\",\"hidden\":true},{\"name\":\"password-confirmation\",\"type\":\"password\",\"placeholder\":\"Confirm\",\"hidden\":true}],\"sequential-params\":true,\"scope\":[\"personal-chats\",\"anonymous\",\"dapps\"],\"scope-bitmask\":42},\"grant-permissions,58\":{\"name\":\"grant-permissions\",\"description\":\"Grant permissions\",\"has-handler\":true,\"async-handler\":false,\"color\":\"#7099e6\",\"icon\":\"lock_white\",\"params\":[],\"execute-immediately?\":true,\"scope\":[\"personal-chats\",\"anonymous\",\"registered\",\"dapps\"],\"scope-bitmask\":58}},\"functions\":{},\"subscriptions\":{}}"}
02-14 13:31:33.428 13266 13315 D StatusModule: endParseJail
02-14 13:31:33.433 28304 28304 I GoogleInputMethod: onFinishInput() : Dummy InputConnection bound
02-14 13:31:33.433 28304 28304 I GoogleInputMethod: onStartInput() : Dummy InputConnection bound
02-14 13:31:33.451 13266 13315 D StatusModule: startNode
02-14 13:31:33.457   564  6410 E msm8974_platform: ramp_speaker_gain: Could not get ctl for mixer cmd - Left Speaker Gain or Right Speaker Gain, not applying speaker gain ramp
02-14 13:31:33.457   564  6410 D audio_hw_primary: out_write: retry previous failed cal level set
02-14 13:31:33.457   564  6410 W msm8974_platform: platform_send_gain_dep_cal: Usecase list is empty
02-14 13:31:33.463 13266 13378 D StatusModule: Starting Geth node in folder: /data/user/0/im.status.ethereum/ethereum/testnet
02-14 13:31:33.506 13266 13378 D StatusModule: Attach to geth.log to instabug /storage/emulated/0/Download/geth.log
02-14 13:31:33.507 13266 13378 D ExtDirLog: /storage/emulated/0/Download/geth.log
```

File: src/status_im/core.cljs [hunk 1]
```diff
@@ -12,8 +12,11 @@
   (object/set js/console "ignoredYellowBox" #js ["re-frame: overwriting"]))
 
 (defn init [app-root]
-  (log/set-level! config/log-level)
+  #_(log/debug "[INIT] log/set-level!")
```
Comment (goranjovic) [inline src/status_im/core.cljs:15]: cleanup comments?

**Review by goranjovic — APPROVED** [review]

**Pull request closed without merging**